library-best-practices/JLBP-5.md

```diff
@@ -36,8 +36,14 @@ Example 2: There are multiple artifacts that provide classes under
     might be used anywhere that is not a servlet. Define these status codes
     yourself or choose a different library to provide them.
 
-Problems caused by overlapping classes can be extremely difficult to resolve.
-If at all possible, eliminate all but one of the
-overlapping dependencies. Otherwise pay very close attention to which version of
-each overlapping class is chosen. Make sure the project does not depend on any
-behavior or API of the class that is not selected.
+In Java 9 and later overlapping classes become compile-time and runtime errors when
+named modules are used. It is critical, especially in Java 9 and later,
+to remove all but one of the overlapping classes from the classpath.
+
+If possible, upgrade all dependencies on the overlapping classes
+to import them only by the same group ID and artifact ID.
+
+If this isn't possible, for instance because a dependency that imports the
+renamed artifact is unmaintained, then add
```

Comment:
"the renamed artifact" doesn't seem to have an antecedent. Also, "dependency" here is confusing because it's a different part of the dependency problem than the part having overlapping classes, unlike the previous paragraph.

I would suggest: "... for instance because you depend on an unmaintained library that imports an undesired artifact containing overlapping classes, ..."

Reply:
rewritten